spec/jobs/deposit_job_spec.rb

```diff
@@ -167,6 +167,35 @@
           expect(external_identifier).to eq('https://cocina.sul.stanford.edu/file/123-456-789')
         end
       end
+
+      context 'when file description has changed' do
+        let(:attached_file_updated) { build(:attached_file, label: 'My changed label') }
+        let(:second_work_version_metadata_only) do
+          build(:work_version, work:, attached_files: [attached_file_updated], version: 2,
+                               version_description: 'Updated metadata')
+        end
+
+        before do
+          work.work_versions = [first_work_version, second_work_version_metadata_only]
+          allow(SdrClient::Find).to receive(:run).and_return(cocina.to_json)
+          allow(attached_file_updated).to receive_message_chain(:file, :blob).and_return(blob) # rubocop:disable RSpec/MessageChain
```

Comment:
I'm not sure why it should be necessary to mock getting a blob back in this test. I don't want to upload new blobs for this metadata-only second version. The goal is to get a test of changing the label on an otherwise unchanged file in a new version, and then add expectations that the size and publish/shelve match the returned cocina from line 90.

Reply:
Worked with Justin to set up a new factory trait to generate a preserved blob instead.